fairworkflows/fairstep.py

```diff
@@ -494,16 +494,26 @@ def _extract_outputs_from_function(func, additional_params) -> List[FairVariable
                          'FAIR step functions MUST have type hinting, '
                          'see https://docs.python.org/3/library/typing.html')
     if _is_generic_tuple(return_annotation):
+        return_sem_types = additional_params.get('out')
+        if return_sem_types is not None:
+            num_return_args = len(return_annotation.__args__)
+            if len(return_sem_types) != num_return_args:
+                raise ValueError(f'"out" parameter to is_fairstep decorator must be a '
+                                  'tuple of length number of returned values (in this case, '
+                                  '{num_return_args}).')
+        else:
+            return_sem_types = [None for arg in return_annotation.__args__]
+
         return [FairVariable(
                     name='out' + str(i + 1),
                     computational_type=annotation.__name__,
-                    semantic_types=additional_params.get('out' + str(i + 1)))
+                    semantic_types=return_sem_types[i])
                 for i, annotation in enumerate(return_annotation.__args__)]
     else:
         return [FairVariable(
                 name='out1',
                 computational_type=return_annotation.__name__,
-                semantic_types=additional_params.get('out1'))]
+                semantic_types=additional_params.get('out'))]
```

Comment:
In this case we should also check (or better: make use of the same check) that the length of 'out' equals the number of output arguments right?

Reply:
Indeed. I have a check further up currently that looks like:
```python
        return_sem_types = additional_params.get('out')
        if return_sem_types is not None:
            num_return_args = len(return_annotation.__args__)
            if len(return_sem_types) != num_return_args:
                raise ValueError(f'"out" parameter to is_fairstep decorator must be a '
                                  'tuple of length number of returned values (in this case, '
                                  '{num_return_args}).')
        else:
            return_sem_types = [None for arg in return_annotation.__args__]
```
I think this should be catching cases where the out tuple is not the right length. Or did you mean something else?

Reply:
Yes that is what I ment.